scripts/lib/CIME/XML/generic_xml.py

```diff
@@ -3,20 +3,42 @@
 be used by other XML interface modules and not directly.
 """
 from CIME.XML.standard_module_setup import *
+import xml.etree.ElementTree as ET
 from distutils.spawn import find_executable
 import getpass
 import six
+from copy import deepcopy
 
 logger = logging.getLogger(__name__)
 
+class _Element(object): # private class, don't want users constructing directly or calling methods on it
```

Comment:
Important: Return nodes to clients as this class means they can no longer make calls directly on nodes; it forces them to go through GenericXML for everything. This can be a little more tedious for clients:

node.text     vs.   db.text(node)

But this is the most-flexible API for us because it makes the fewest assumptions.